.github/workflows/build.yml

```diff
@@ -68,13 +67,37 @@ jobs:
       - name: publish image to dockerhub
         run: |
           SAMPLE_LIST=(hello-kubernetes distributed-calculator pub-sub bindings observability secretstore)
-
           for sample in "${SAMPLE_LIST[@]}"; do
             echo "Push image for ${sample}..."
             pushd ${sample}
             make push
             popd
           done
+      - name: GitHub container registry login
+        if: matrix.target_os != 'darwin' && github.event_name != 'pull_request'
```

Comment:
Nit, but do we need this OS check?

Reply:
Yes. We don't want to publish the image per OS build.